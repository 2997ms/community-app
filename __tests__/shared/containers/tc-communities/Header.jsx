import _ from 'lodash';
import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';

const rnd = new Rnd();

const mockHeaderActions = {
  tcCommunities: {
    header: {
      fetchDataInit: jest.fn(),
      fetchDataDone: jest.fn(),
      mobileToggle: jest.fn(),
    },
  },
};
jest.setMock(require.resolve('actions/tc-communities/meta'), mockHeaderActions);

const mockState = {
  tcCommunities: {
    meta: {
      communityId: 'someId',
      loading: false,
      menuItems: [
        { title: 'Menu Item 1', url: 'pageId1' },
        { title: 'Menu Item 2', url: 'pageId2' },
        { title: 'Menu Item 3', url: 'pageId3' },
      ],
      logos: ['some/logo/url'],
      cssUrl: 'some/css/url',
      isMobileOpen: false,
      failed: false,
    },
  },
};

const mockState2 = {
  tcCommunities: {
    meta: {},
  },
};

const mockState3 = {
  tcCommunities: {
    meta: {
      communityId: 'anotherId',
    },
  },
};

const mockState4 = {
  tcCommunities: {
    meta: {
      communityId: 'someId',
      isMobileOpen: true,
    },
  },
};

const Header = require('containers/tc-communities/Header').default;

beforeEach(() => jest.clearAllMocks());

test('Matches shapshot', () => {
  rnd.render((
    <Header
      match={{
        params: {
          communityId: 'someId',
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState,
        subscribe: _.noop,
      }}
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});

test('Triggers data loading if not loaded yet', () => {
  TU.renderIntoDocument((
    <Header
      match={{
        params: {
          communityId: 'someId',
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState2,
        subscribe: _.noop,
      }}
    />
  ));
  expect(mockHeaderActions.tcCommunities.header.fetchDataInit).toHaveBeenCalled();
  expect(mockHeaderActions.tcCommunities.header.fetchDataDone).toHaveBeenCalledWith('someId');
});

test('Triggers data loading if loaded for another community', () => {
  TU.renderIntoDocument((
    <Header
      match={{
        params: {
          communityId: 'someId',
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState3,
        subscribe: _.noop,
      }}
    />
  ));
  expect(mockHeaderActions.tcCommunities.header.fetchDataInit).toHaveBeenCalled();
  expect(mockHeaderActions.tcCommunities.header.fetchDataDone).toHaveBeenCalledWith('someId');
});

test('Triggers toggle mobile menu on click', () => {
  const page = TU.renderIntoDocument((
    <Header
      match={{
        params: {
          communityId: 'someId',
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState3,
        subscribe: _.noop,
      }}
    />
  ));

  const btn = TU.findAllInRenderedTree(page, item =>
    item && item.className && item.className.match(/mobile-toggle/));
  expect(btn.length).toBe(1);
  TU.Simulate.click(btn[0]);
  expect(mockHeaderActions.tcCommunities.header.mobileToggle).toHaveBeenCalled();
});

test('Close mobile menu when mount', () => {
  TU.renderIntoDocument((
    <Header
      match={{
        params: {
          communityId: 'someId',
        },
      }}
      store={{
        dispatch: () => _.noop,
        getState: () => mockState4,
        subscribe: _.noop,
      }}
    />
  ));

  expect(mockHeaderActions.tcCommunities.header.mobileToggle).toHaveBeenCalled();
});
