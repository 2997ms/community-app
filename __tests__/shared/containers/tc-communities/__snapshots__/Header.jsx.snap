// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shapshot 1`] = `
<HeaderContainer
  communityId="someId"
  cssUrl="some/css/url"
  isCommunityNotFound={false}
  isLoading={false}
  isMobileOpen={false}
  loadData={[Function]}
  loadedCommunityId="someId"
  logos={
    Array [
      "some/logo/url",
    ]
  }
  match={
    Object {
      "params": Object {
        "communityId": "someId",
      },
    }
  }
  menuItems={
    Array [
      Object {
        "title": "Menu Item 1",
        "url": "pageId1",
      },
      Object {
        "title": "Menu Item 2",
        "url": "pageId2",
      },
      Object {
        "title": "Menu Item 3",
        "url": "pageId3",
      },
    ]
  }
  mobileToggle={[Function]}
  profile={null}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
